compiler/backend/src/org/jetbrains/kotlin/codegen/MemberCodegen.java

```diff
@@ -313,14 +315,21 @@ private boolean shouldInitializeProperty(@NotNull JetProperty property) {
         PropertyDescriptor propertyDescriptor = (PropertyDescriptor) bindingContext.get(VARIABLE, property);
         assert propertyDescriptor != null;
 
-        CompileTimeConstant<?> compileTimeValue = propertyDescriptor.getCompileTimeInitializer();
+        JetExpression initializer = property.getInitializer();
+
+        CompileTimeConstant<?> initializerValue =
+                property.isVar() && initializer != null
+                ? ConstantExpressionEvaluator.OBJECT$.evaluate(initializer, state.getBindingTrace(), null)
```

Comment:
I suggest to use a temporary trace here (and discard it) instead of `state.getBindingTrace()` so that `ConstantExpressionEvaluator` won't unexpectedly write to the global trace and alter the subsequent code generation.